streams/order-data.go

```diff
@@ -0,0 +1,35 @@
+package streams
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
+type OrderData struct {
+	Rate   float64
+	Type   string
+	Amount float64
+}
+
+func (od *OrderData) UnmarshalJSON(b []byte) error {
+	var rawStrings map[string]string
```

Comment:
Please unmarshall directly into the object. I'm on the phone right now. So I'm not able to write correct go-code. But google it!